src/App.jsx

```diff
@@ -14,45 +15,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
-
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+const SORT_FIELD_ALPH = 'alph';
+const SORT_BY_LENGTH = 'length';
+const SORT_REVERSE = 'reverse';
+
+function getSortedGoods(goods, { sortField, direction }) {
+  const sortedGoods = [...goods];
+
+  if (sortField) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPH:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (direction) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [direction, setDirection] = useState('');
+  const sortedGoods = getSortedGoods(goodsFromServer, { sortField, direction });
+
+  return (
+    <>
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${sortField !== SORT_FIELD_ALPH && 'is-light'}`}
+            onClick={() => setSortField(SORT_FIELD_ALPH)}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-success ${sortField !== SORT_BY_LENGTH && 'is-light'}`}
+            onClick={() => setSortField(SORT_BY_LENGTH)}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-warning ${direction !== SORT_REVERSE && 'is-light'}`}
+            onClick={() => setDirection(!direction ? SORT_REVERSE : '')}
+          >
+            Reverse
+          </button>
+
+          {(sortField !== '' || direction !== '')
+            && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                setSortField('');
+                setDirection('');
+              }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here